src/cloudant/database.py

```diff
@@ -160,7 +160,14 @@ def create_document(self, data, throw_on_exists=False):
         if throw_on_exists and doc.exists():
             raise CloudantDatabaseException(409, docid)
         doc.update(data)
-        doc.create()
+        try:
+            doc.create()
+        except HTTPError as error:
+            if error.response.status_code == 409:
+                if throw_on_exists:
+                    raise
```

Comment:
I thought that you would remove lines 160 and 161 and replace this line 168 with a `raise CloudantDatabaseException(409, docid)`.  Shouldn't that really do the trick to handle both conditions here?